setup.py

```diff
@@ -28,20 +47,62 @@ def proto_compile(output_path):
             file.truncate()
 
 
-class ProtoCompileInstall(install):
+def libp2p_build_install():
+    try:
+        proc = subprocess.Popen(['go', 'version'],
+                                stdout=subprocess.PIPE)
+        result, _ = proc.communicate()
+        result = result.decode('ascii', 'replace')
+        _, _, v, _ = result.split(' ')
```

Comment:
This method of parsing output is not very readable and reliable. I suggest to use the regular expression like `^go version go([\d.]+)` instead.